components/raftstore/src/store/fsm/apply.rs

```diff
@@ -2444,7 +2444,7 @@ impl Debug for GenSnapTask {
 static OBSERVE_ID_ALLOC: AtomicUsize = AtomicUsize::new(0);
 
 /// A unique identifier for checking stale observed commands.
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
```

Comment:
We need to use `Ordering::SeqCst` to new `ObserveID` if we use the value to determine staleness between different `ObserveID`s.

Reply:
I don't think so, all the updates and comparisons would happen in a single thread.

Reply:
But still, it is a pitfall for futures improvement (eg, multi-thread CDC).